### PR TITLE
Python 3

### PR DESCRIPTION
**Pull request by njwilson23**
I'm working on setting up single-codebase Python 3 compatibility. Is this something that could eventually be merged upstream?

### CODE REVIEW

**Comment by scisco** [comment]
@njwilson23 absolutely. I've been looking to this myself. Thanks for doing this. Will take a look and merge

File: landsat/uploader.py [hunk 1]
```diff
@@ -11,16 +11,20 @@
...
+except ImportError:
+    import Queue as queue
 from multiprocessing import pool
 try:
     import cStringIO
     StringIO = cStringIO
 except ImportError:
-    import StringIO
+    from io import StringIO
+    #import StringIO        # FIXME: why original fall back to StringIO in Py2?
```
Comment (njwilson23) [inline landsat/uploader.py]: The original use StringIO as a fall back if cStringIO failed to import. When does this happen, and is it a concern?
Also, need to look carefully at the return types were io.StringIO is used.

**Comment by njwilson23** [comment]
What is the best way to get testing under different interpreters set up, using the current docker-based tests?

**Comment by scisco** [comment]
@njwilson23 Your pull request still needs more work to properly work on python 3. I'll close this for now until further work is done here. There is another ticket open on this here: https://github.com/developmentseed/landsat-util/issues/170

I'm also reworking the CI tests, removing docker so we can test python 2 and 3

**Comment by njwilson23** [comment]
Sounds good - working out testing was the obstacle to identifying the problems and finishing this up.

**Comment by scisco** [comment]
@njwilson23 we added python 3.0 support in `0.13.0`. Check it out! Thanks for your help.

**Comment by njwilson23** [comment]
Awesome! Hope to be able to help on other things in the future.

**Pull request closed without merging**